lib/ip-locator.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const Ipinfo = require('ipinfo')
+const RequestIp = require('request-ip')
+
+class IpLocator {
+  constructor (options) {
+    const { enabledByDefault = true } = options
+    this.enabledByDefault = enabledByDefault
+  }
+
+  async handle (request, h) {
+    if (this.isDisabledFor(request)) {
+      return h.continue
+    }
+    request.location = await this.locate(request)
+
+    return h.continue
+  }
+
+  routeConfig (request) {
+    return request.route.settings.plugins['hapi-geo-locate'] || {}
+  }
+
+  isDisabledFor (request) {
+    return !this.isEnabledFor(request)
+  }
+
+  isEnabledFor (request) {
+    const { enabled = this.enabledByDefault } = this.routeConfig(request)
+
+    return enabled
+  }
+
+  locate (request) {
+    const ip = this.ipFrom(request)
+
+    return new Promise((resolve, reject) => {
+      Ipinfo(ip, (err, location) => {
```

Comment:
Can you please use the implementation for Ipinfo from the master branch? It simplifies the handling from callbacks to async/await

Reply:
ohk sir